lib/CMakeLists.txt

```diff
@@ -1,10 +1,17 @@
-cmake_minimum_required(VERSION 3.12)
-
-project(vmcontainer)
```

Comment:
Why did you have to remove these? Does this still support the use case of consuming the library directly through `include(lib/CMakeLists.txt)`?

Reply:
Oh, one would need to do `add_subdirectory(vmcontainer)` where `vmcontainer` is the root of this project. You need this since when you embed a project into another in CMake, you need to export the targets or users won't be able to install (or export) their own targets. This is not a usecase I had in mind, I'll do some tests on my side to allow this.

Reply:
> `add_subdirectory(vmcontainer)`

That's what I meant, sorry, got them confused. The goal here was that users can do `add_subdirectory(vmcontainer/lib)` and only get the library without the benchmarks and tests.

Reply:
I modified the top level CMake file to support being embedded in other projects using `add_subdirectory(vmcontainer)`. I also put everything related to installation to the `lib/CMakeLists.txt` to be more consistent.